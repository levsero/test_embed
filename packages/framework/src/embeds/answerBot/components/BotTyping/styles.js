import styled from 'styled-components'
import { LoadingEllipses } from 'component/loading/LoadingEllipses'
import { zdColorGrey600, zdColorGrey100 } from '@zendeskgarden/css-variables'

const Container = styled.div`
  ${(props) => {
    return `
      display: inline-block;
      max-width: 100%;
      float: ${props.theme.rtl ? 'right' : 'left'};
      background: ${zdColorGrey100};
      border-radius: ${16 / props.theme.fontSize}rem;
      padding: ${8 / props.theme.fontSize}rem ${12 / props.theme.fontSize}rem;
      ${props.theme.rtl ? 'margin-right' : 'margin-left'}: ${40 / props.theme.fontSize}rem;
    `
  }}
`

const Loader = styled(LoadingEllipses)`
  ${(props) => {
    return `
      margin: ${4 / props.theme.fontSize}rem;

      div {
        width: ${5.5 / props.theme.fontSize}rem;
        height: ${5.5 / props.theme.fontSize}rem;
        background-color: ${zdColorGrey600};
        ${props.theme.rtl ? 'margin-left' : 'margin-right'}: ${4 / props.theme.fontSize}rem;
      }
    `
  }}
`

export { Container, Loader }
