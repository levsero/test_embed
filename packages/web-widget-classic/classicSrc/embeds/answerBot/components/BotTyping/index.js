import { TEST_IDS } from 'classicSrc/constants/shared'
import { Container, Loader } from './styles'

const BotTyping = () => (
  <Container data-testid={TEST_IDS.AB_TYPING_INDICATOR}>
    <Loader useUserColor={false} />
  </Container>
)

export default BotTyping
