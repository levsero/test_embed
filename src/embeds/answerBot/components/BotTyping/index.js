import React from 'react'

import { Container, Loader } from './styles'
import { TEST_IDS } from 'constants/shared'

const BotTyping = () => (
  <Container data-testid={TEST_IDS.AB_TYPING_INDICATOR}>
    <Loader useUserColor={false} />
  </Container>
)

export default BotTyping
