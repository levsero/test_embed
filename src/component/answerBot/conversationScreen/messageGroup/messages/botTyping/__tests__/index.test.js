import { render } from 'react-testing-library';
import React from 'react';

import BotTyping from '../index';

test('renders the expected classes', () => {
  const { container } = render(<BotTyping />);

  expect(container)
    .toMatchSnapshot();
});
