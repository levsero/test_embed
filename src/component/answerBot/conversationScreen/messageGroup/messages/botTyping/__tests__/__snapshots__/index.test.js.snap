// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected classes 1`] = `
<div>
  <div
    class="bubble"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="container loadingContainer"
      data-testid="Icon--ellipsis"
    >
      <div
        class="
      circle loadingAnimation
      bounce
      
    "
      />
      <div
        class="
      circle loadingAnimation
      bounce
      
    "
      />
      <div
        class="
      circle loadingAnimation
      bounce
      
    "
      />
    </div>
  </div>
</div>
`;
